rust.003.vectorized/src/main.rs

```diff
@@ -23,13 +123,22 @@ fn main() {
             continue;
         }
 
-        // G/C & 10 == 2
-        // A/T & 10 == 0
-        // N & 10 == 10
-        for c in line.iter() {
-            gc += (*c & 10 == 2) as u32;
-            at += (*c & 10 == 0) as u32;
-        }
+        // Try to read more lines
```

Comment:
This reads more lines into the buffer if the lines aren't header lines, up to BUFSIZE. I think this still fits the "read line by line" rule since it is still reading up to line breaks, it's just delaying the processing of the extra lines till a modest buffer size has been hit to allow the simd code go to work on larger chunks at a time.